grails-app/services/org/pih/warehouse/inventory/CycleCountService.groovy

```diff
@@ -169,4 +173,94 @@ class CycleCountService {
 
         return csv
     }
+
+    /**
+     * Batch method to start a cycle count.
+     * Fetch associated Cycle count with the CycleCountRequest and do the following:
+     * if a cycle count doesn't exist, create it, fetch all bin locations + inventory items for the product in a request,
+     * get the product availability for each item, filter out items that have qoh = 0 (but keep those that have qoh = 0 and were included in the previous count)
+     *
+     * */
+    List<CycleCountDto> startCycleCount(CycleCountStartBatchCommand command) {
+        List<CycleCountDto> cycleCounts = []
+        command.requests.each { CycleCountStartCommand request ->
+            CycleCount cycleCount = request.cycleCountRequest.cycleCount
+            if (!cycleCount) {
+                CycleCountDto cycleCountDto = createCycleCount(request, command.facility)
+                cycleCounts.add(cycleCountDto)
+                return
+            }
+            List<CycleCountItem> cycleCountItems = CycleCountItem.findAllByCycleCount(cycleCount)
+            CycleCountDto cycleCountDto = updateCycleCount(cycleCount, cycleCountItems, request)
+            cycleCounts.add(cycleCountDto)
+        }
+        return cycleCounts
+    }
+
+    CycleCountDto createCycleCount(CycleCountStartCommand request, Location facility) {
+        CycleCount newCycleCount = new CycleCount(
+                facility: facility,
+                dateLastRefreshed: new Date()
+        )
+        List<AvailableItem> itemsToSave = determineCycleCountItemsToSave(facility, request.cycleCountRequest.product)
+        newCycleCount.save()
+        // 1:1 association between cycle count and cycle count request
+        request.cycleCountRequest.cycleCount = newCycleCount
+        List<CycleCountItem> cycleCountItems = []
+        itemsToSave.each { AvailableItem availableItem ->
+            CycleCountItem cycleCountItem = new CycleCountItem(
+                    status: CycleCountItemStatus.READY_TO_COUNT,
+                    countIndex: request.countIndex,
+                    quantityOnHand: availableItem.quantityOnHand,
+                    quantityCounted: availableItem.quantityOnHand == 0 ? 0 : null,
+                    cycleCount: newCycleCount,
+                    facility: facility,
+                    inventoryItem: availableItem.inventoryItem,
+                    product: availableItem.inventoryItem.product,
+                    createdBy: AuthService.currentUser,
+                    updatedBy: AuthService.currentUser,
+                    // FIXME: I don't know what default values for custom and draft should be
+                    custom: false,
+                    draft: true,
```

Comment:
as mentioned in the FIXME - I am not sure what default values for those should be, as they weren't determined anywhere in the docs.

Reply:
tech huddle: remove the draft property